mteb/leaderboard/table.py

```diff
@@ -138,34 +139,43 @@ def scores_to_tables(
     joint_table.insert(1, "mean_by_task_type", typed_mean)
     joint_table["borda_rank"] = get_borda_rank(per_task)
     joint_table = joint_table.reset_index()
-    joint_table = joint_table.drop(columns=["model_revision"])
     model_metas = joint_table["model_name"].map(failsafe_get_model_meta)
     joint_table = joint_table[model_metas.notna()]
     joint_table["model_link"] = model_metas.map(lambda m: m.reference)
     joint_table.insert(
         1,
         "Max Tokens",
-        model_metas.map(lambda m: str(int(m.max_tokens)) if m.max_tokens else ""),
+        model_metas.map(
+            lambda m: str(int(m.max_tokens)) if m.max_tokens else "Unknown"
+        ),
     )
     joint_table.insert(
         1,
         "Embedding Dimensions",
-        model_metas.map(lambda m: str(int(m.embed_dim)) if m.embed_dim else ""),
+        model_metas.map(lambda m: str(int(m.embed_dim)) if m.embed_dim else "Unknown"),
     )
     joint_table.insert(
         1,
         "Number of Parameters",
         model_metas.map(lambda m: format_n_parameters(m.n_parameters)),
     )
     joint_table = joint_table.sort_values("borda_rank", ascending=True)
+    per_task = per_task.loc[
+        joint_table.set_index(["model_name", "model_revision"]).index
+    ]
+    joint_table = joint_table.drop(columns=["model_revision"])
     # Removing HF organization from model
     joint_table["model_name"] = joint_table["model_name"].map(
         lambda name: name.split("/")[-1]
     )
     # Adding markdown link to model names
-    joint_table["model_name"] = (
-        "[" + joint_table["model_name"] + "](" + joint_table.pop("model_link") + ")"
+    name_w_link = (
+        "[" + joint_table["model_name"] + "](" + joint_table["model_link"] + ")"
     )
```

Comment:
Non-blocking: I would probably just create the column initially as model_name_w_link using a method on the model_meta object like model_meta.model_name_as_markdown(). 

(could imagine we could use this elsewhere and also removed completely from the leaderboard code)

Reply:
Yeah, seems reasonable, I think I will do this in a future PR